src/main/java/com/multithreading/TextProcessor.java

```diff
@@ -0,0 +1,38 @@
+package com.multithreading;
+
+class Capitalize implements Runnable {
+    private final char[] chars;
```

Comment:
although it technically works, you are not really solving the problem in the spirit of the problem since each thread has a full copy of the string instead of just managing a single character. I would expect to see each thread only aware of a single character. The benefit also is you won't need to synchronize anything.